sycl/include/sycl/detail/optional.hpp

```diff
@@ -63,15 +63,15 @@ template <typename T> class optional {
   template <typename U> optional &operator=(const optional<U> &Other) {
     if (has_value())
       reinterpret_cast<T *>(Storage)->~T();
-    ContainsValue = Other;
-    new (Storage) T(Other.Value);
+    ContainsValue = bool(Other);
```

Comment:
If you want to be explicit about it, I'd prefer this:
```suggestion
    ContainsValue = Other.has_value();
```
Just a nitpick though.